### ISSUE TITLE
Naming return values (?)

### ISSUE THREAD

**Issue by st0012** [goby-lang/goby#558]
According to [this article](https://blog.minio.io/golang-internals-part-2-nice-benefits-of-named-return-values-1e95305c8687), we can get smaller and more efficient binary code generated if we name function's return values.

So I think it'll be great for Goby if we can all do this in the future and refactoring the existing code. @hachi8833 @saveriomiroddi

**Comment by saveriomiroddi** [comment]
There is a very interesting discussion in the related [github issue](https://github.com/golang/go/issues/20859).

**Comment by st0012** [comment]
Yep, I've read them. Actually, except the compilation optimization we can get, I also think it'll help us on code's readability. I think at this point our codebase is lack of documentation, naming return values should help that. 
And I know there's a potential risk about shadowing variables, but I think it can be avoid if we use proper [IDE](https://www.jetbrains.com/go/) or editors.
